pkg/helm/charts/tests/securedclusterservices/flavor/flavor_test.go

```diff
@@ -32,27 +33,23 @@ func TestOverriddenTagsAreRenderedInTheChart(t *testing.T) {
 
 func TestWithDifferentImageFlavors(t *testing.T) {
 	testbuildinfo.SetForTest(t)
-	// having a function as value allows to successfully run this test without dependency to GOTAGS='' and GOTAGS='release'
-	imageFlavorCases := map[string]func() defaults.ImageFlavor{
-		"development": func() defaults.ImageFlavor {
-			testutils.SetVersion(t, testutils.GetExampleVersion(t))
-			return defaults.DevelopmentBuildImageFlavor()
-		},
-		"stackrox": func() defaults.ImageFlavor {
-			testutils.SetVersion(t, testutils.GetExampleVersionUnified(t))
-			return defaults.StackRoxIOReleaseImageFlavor()
-		},
-		"rhacs": func() defaults.ImageFlavor {
-			testutils.SetVersion(t, testutils.GetExampleVersionUnified(t))
-			return defaults.RHACSReleaseImageFlavor()
-		},
+	testutils.SetVersion(t, testutils.GetExampleVersion(t))
+	imageFlavorCases := map[string]defaults.ImageFlavor{
+		"development": defaults.DevelopmentBuildImageFlavor(),
+		"stackrox":    defaults.StackRoxIOReleaseImageFlavor(),
+		"rhacs":       defaults.RHACSReleaseImageFlavor(),
+	}
+	if buildinfo.ReleaseBuild {
+		imageFlavorCases["opensource-release"] = defaults.OpenSourceImageFlavor()
+	} else {
+		imageFlavorCases["opensource-development"] = defaults.OpenSourceImageFlavor()
```

Comment:
Nice! I like how you split to run the respective `yaml` file based on the build flag.

Reply:
This was a workaround to me not seeing any elegant way to express the condition within one file :-P